### PR TITLE
Test: Driver can control connection mode

### PR DESCRIPTION
**Pull request by anthony-murphy**
Adds a test the ensure the driver can override the connection mode regardless of what the client requests and removes an assert that is no longer needed or valid.

[AB#8194](https://dev.azure.com/fluidframework/internal/_workitems/edit/8194)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/driverConnectionMode.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+				},
+			);
+
+			const overrideContainer = await loader.resolve({ url: await getExistingContainerUrl });
+
+			await waitForContainerConnection(overrideContainer, true);
+
+			assert.strictEqual(
+				overrideContainer.deltaManager.active,
+				driverConnectionMode === "write",
```
Comment (dannimad) [inline packages/test/test-end-to-end-tests/src/test/driverConnectionMode.spec.ts:63]: What's the expected behavior when driverConnectionMode === "read" and forceWriteConnection === true ?

Comment (anthony-murphy) [reply packages/test/test-end-to-end-tests/src/test/driverConnectionMode.spec.ts:63]: the expected mode is "read". The driver/server owns the mode, the client simply requests a mode.

**Review by dannimad — APPROVED** [review]
Config flag should be a bit clearer imo. Instead of 'forceWriteConnection' be something like 'requestWriteConnection' but otherwise LGTM

**Comment by anthony-murphy** [comment]
> Config flag should be a bit clearer imo. Instead of 'forceWriteConnection' be something like 'requestWriteConnection' but otherwise LGTM

yeah, but i don't want to change it, as its in use in a number of places. hopefully this test helps make it clear

**Pull request merged**